apps/prairielearn/assets/scripts/lib/questionsTable.js

```diff
@@ -0,0 +1,194 @@
+/* eslint-env browser, jquery */
+
+import _ from 'lodash';
+import { onDocumentReady, decodeData } from '@prairielearn/browser-utils';
+import { html } from '@prairielearn/html';
+
+onDocumentReady(() => {
+  const { course_instance_ids, showAddQuestionButton, urlPrefix, plainUrlPrefix } =
+    decodeData('questions-table-data');
+  window.topicList = function () {
+    var data = $('#questionsTable').bootstrapTable('getData');
+    return _.keyBy(_.map(data, (row) => row.topic.name));
+  };
+
+  window.tagsList = function () {
+    var data = $('#questionsTable').bootstrapTable('getData');
+    return _.keyBy(_.map(_.flatten(_.filter(_.map(data, (row) => row.tags))), (row) => row.name));
+  };
+
+  window.versionList = function () {
+    var data = $('#questionsTable').bootstrapTable('getData');
+    return _.keyBy(_.map(data, (row) => row.display_type));
+  };
+
+  window.qidFormatter = function (qid, question) {
+    var text = '';
+    if (question.sync_errors) {
+      text += html`<button
+        class="btn btn-xs mr-1"
+        data-toggle="popover"
+        data-title="Sync Errors"
+        data-content='<pre style="background-color: black" class="text-white rounded p-3">${_.escape(
+          question.sync_errors_ansified,
+        )}</pre>'
+      >
+        <i class="fa fa-times text-danger" aria-hidden="true"></i>
+      </button>`;
+    } else if (question.sync_warnings) {
+      text += html`<button
+        class="btn btn-xs mr-1"
+        data-toggle="popover"
+        data-title="Sync Warnings"
+        data-content='<pre style="background-color: black" class="text-white rounded p-3">${_.escape(
+          question.sync_warnings_ansified,
+        )}</pre>'
+      >
+        <i class="fa fa-exclamation-triangle text-warning" aria-hidden="true"></i>
+      </button>`;
+    }
+    text += html`<a class="formatter-data" href="${urlPrefix}/question/${question.id}/"
+      >${_.escape(question.qid)}</a
+    >`;
+    if (question.open_issue_count > 0) {
+      text += html`<a
+        class="badge badge-pill badge-danger ml-1"
+        href="${urlPrefix}/course_admin/issues?q=is%3Aopen+qid%3A${_.escape(question.qid)}"
+        >${question.open_issue_count}</a
+      >`;
+    }
+    return text.toString();
+  };
+
+  window.topicFormatter = function (topic, question) {
+    return html`<span class="badge color-${question.topic.color}"
+      >${_.escape(question.topic.name)}</span
+    >`.toString();
+  };
+
+  window.tagsFormatter = function (tags, question) {
+    return _.map(question.tags ?? [], (tag) =>
+      html`<span class="badge color-${tag.color}">${tag.name}</span>`.toString(),
+    ).join(' ');
+  };
+
+  window.versionFormatter = function (version, question) {
+    return html`<span class="badge color-${question.display_type === 'v3' ? 'green1' : 'red1'}"
+      >${_.escape(question.display_type)}</span
```

Comment:
I don't think escape is needed here since you're using the html module...
```suggestion
      >${question.display_type}</span
```

Reply:
done